<?xml version="1.0" encoding="UTF-8"?>
<launch>

  <arg name="camera_info_topic" default="xtion/depth/camera_info"/>
  <arg name="rgb_image_topic" default="xtion/rgb/image_rect_color"/>
  <arg name="camera_frame_id" default="camera_link"/>
  <arg name="global_frame_id" default="map"/>
  <arg name="base_frame_id" default="base_link"/>
  <arg name="camera_pub_frequency" default="20.0"/>

  <param name="robot_description" textfile="$(find adream_simulation)/models/cad_models/urdf/r2d2.urdf"/>

  <node name="camera_publisher" pkg="uwds3_perception" type="camera_publisher_node.py" args="" output="screen" respawn="false">
    <rosparam subst_value="true">
      camera_info_topic: $(arg camera_info_topic)
      rgb_image_topic: $(arg rgb_image_topic)
      camera_frame_id: optical_frame_$(arg camera_frame_id)
      camera_pub_frequency: $(arg camera_pub_frequency)
    </rosparam>
  </node>

  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher"/>
  <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher" args="_use_gui:=true"/>

  <node name="camera_optical_frame_publisher" pkg="tf2_ros" type="static_transform_publisher" args="0 0 0 -1.57 0 -1.57 $(arg camera_frame_id) optical_frame_$(arg camera_frame_id)"/>
  <node name="odom2base_publisher" pkg="tf2_ros" type="static_transform_publisher" args="0 0 0.47 0 0 0 odom $(arg base_frame_id)"/>
  <node name="map2odom_publisher" pkg="tf2_ros" type="static_transform_publisher" args="-5 16 0 1.57 0 0 $(arg global_frame_id) odom"/>

</launch>
